lib/omniauth/strategies/azure_oauth2.rb

```diff
@@ -31,7 +31,6 @@ def client
           provider.respond_to?(:base_azure_url) ? provider.base_azure_url : BASE_AZURE_URL
 
         options.authorize_params.domain_hint = provider.domain_hint if provider.respond_to?(:domain_hint) && provider.domain_hint
-        options.authorize_params.prompt = request.params['prompt'] if request.params['prompt']
```

Comment:
I agree with the premise that getting client shouldn't depend on a request. However, it would be safer to 
`options.authorize_params.prompt = request.params['prompt'] if defined? request && request.params['prompt']`